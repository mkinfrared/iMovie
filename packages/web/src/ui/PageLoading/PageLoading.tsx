import CircularProgress from "@material-ui/core/CircularProgress";
import React from "react";

import css from "./PageLoading.module.scss";

const PageLoading = () => {
  return (
    <div className={css.PageLoading}>
      <CircularProgress color="secondary" />
    </div>
  );
};

export default React.memo(PageLoading);
