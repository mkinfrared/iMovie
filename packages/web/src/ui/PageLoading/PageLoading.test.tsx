import { render } from "@testing-library/react";
import React, { ReactElement } from "react";

import PageLoading from "./PageLoading";

describe("<PageLoading />", () => {
  let Component: ReactElement;

  beforeEach(() => {
    Component = <PageLoading />;
  });

  it("should be defined", () => {
    const { container } = render(Component);

    expect(container).toBeDefined();
  });

  it("should match a snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });
});
