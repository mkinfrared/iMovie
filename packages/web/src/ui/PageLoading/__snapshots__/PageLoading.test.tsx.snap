// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageLoading /> should match a snapshot 1`] = `
<div>
  <div
    class="PageLoading"
  >
    <div
      class="MuiCircularProgress-root MuiCircularProgress-colorSecondary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
</div>
`;
